experimental/Permutations/Permutations.jl

```diff
@@ -0,0 +1,136 @@
+# This file implements a new way to input permutations in Julia. For example
+# it is possible to create a permutation as follow
+# pi = Oscar.Permutations.@perm (1,2,3)(4,5)(6,7,8)
+# > (1,2,3)(4,5)(6,7,8)
+# For this we use macros to modify the syntax tree of (1,2,3)(4,5)(6,7,8) such that
+# Julia can deal with the expression.
+
+module Permutations
+
+ using GAP
+ using Oscar
+
+################################################################################
+# Macro to input a permutation as
+# julia> pi = Oscar.Permutations.@perm (1,2,3)(4,5)(6,7,8)
+# > (1,2,3)(4,5)(6,7,8)
+#
+macro perm(ex)
+    h = (ex).head
+    arg = (ex).args
+    res = []
+    
+    while h != :tuple
+        arg1 = arg[1]
+        insert!(res,1,Expr(:vect,arg[2:length(arg)]...))
+        h = (arg1).head
+        arg = (arg1).args
+    end
```

Comment:
This is a matter of taste I guess, but how about replacing `arg1` by just ex, i.e., write the code more like this:
```julia
  while ex.head == :call
     pushfirst!(res, Expr(:vect, ex.args[2:end]...))
     ex = ex.args[1]
  end
```